client/state/login/reducer.js

```diff
@@ -56,6 +59,9 @@ export const redirectTo = createReducer( null, {
 	[ SOCIAL_CONNECT_ACCOUNT_REQUEST ]: () => null,
 	[ SOCIAL_CONNECT_ACCOUNT_REQUEST_FAILURE ]: ( state, action ) => get( action, 'redirectTo', null ),
 	[ SOCIAL_CONNECT_ACCOUNT_REQUEST_SUCCESS ]: () => null,
+	[ LOGOUT_REQUEST ]: () => null,
+	[ LOGOUT_REQUEST_SUCCESS ]: () => ( state, { data } ) => get( data, 'redirect_to', null ),
+	[ LOGOUT_REQUEST_FAILURE ]: () => null,
 } );
```

Comment:
Shouldn't we reset the whole `login` subtree upon a successful logout?

Right now we only clean up `login.redirectTo`.